conda/gateways/connection/session.py

```diff
@@ -182,8 +203,6 @@ def __init__(self, auth: AuthBase | tuple[str, str] | None = None):
 
         self.headers["User-Agent"] = context.user_agent
 
-        self.verify = context.ssl_verify
-
```

Comment:
Something is wrong here with the `verify` boolean, we stop setting it here but above it's not being set if `truststore` is configured but doesn't work for some reason.

I think the fallback should not be the default of the session parent class but bubbling up as an error to the user since it may lead to surprising results.

E.g. if a user has `truststore` configured explicitly, they should be told the reason in case it doesn't work, since the non-OS certs may not work for their use case after all.

Reply:
I changed it so it now raises a CondaError with a message stating that truststore is only available on python 3.10 or above.

I also made sure `self.verify = True` when `truststore` to ensure requests validates the certificate.